packages/overture-schema-system/src/overture/schema/system/json_schema.py

```diff
@@ -0,0 +1,204 @@
+"""
+JSON Schemas of Overture-based Pydantic models.
+"""
+
+from enum import Enum
+from types import UnionType
+from typing import Annotated, Any, Union, cast, get_args, get_origin
+
+from pydantic import BaseModel, TypeAdapter
+from pydantic.json_schema import GenerateJsonSchema, JsonSchemaValue
+from pydantic_core import core_schema
+from typing_extensions import override
+
+
+class GenerateOmitNullableOptionalJsonSchema(GenerateJsonSchema):
+    """
+    Generates a JSON Schema in which optional values that allow `None` are omitted instead of being
+    assigned a `null` value.
+
+    Example
+    -------
+    >>> import json
+    >>> from pydantic import BaseModel, Field
+    >>>
+    >>> class FooModel(BaseModel):
+    ...     foo: str | None = None
+    ...
+    >>> print(json.dumps(FooModel.model_json_schema(
+    ...     schema_generator=GenerateOmitNullableOptionalJsonSchema
+    ... ), indent=2))
+    {
+      "properties": {
+        "foo": {
+          "title": "Foo",
+          "type": "string"
+        }
+      },
+      "title": "FooModel",
+      "type": "object"
+    }
+
+
+    ⚠️ Warning
+    ----------
+    When using this class to generate a JSON Schema, you must dump your model JSON carefully to
+    ensure that it matches the JSON Schema.
```

Comment:
The need for this warning is why I think `Omitable` is needed.

It's weird for us to say "Overture uses this omit convention but if you use our system you have to think carefully about matching your JSON Schema and JSON dumps of your model"...

Reply:
Sure, but that still feels like a later problem if we find that lots of people are using this with JSON data + JSON Schema (vs. other approaches that are more in line with how our data pipeline is going to use it).